library/net.c

```diff
@@ -751,6 +751,9 @@ int dnet_process_recv(struct dnet_backend_io *backend, struct dnet_net_state *st
 
 void dnet_state_remove_nolock(struct dnet_net_state *st)
 {
+	if (st->__join_state == DNET_JOIN)
```

Comment:
It was: list_for_each_entry(st, &n->dht_state_list, node_entry) {num++;

Why current implementation doesn't use node->storage_state_list?